File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/GrandSubjectDB/GrandSubjectColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { GrandSubjectRow } from "./GrandSubjectRow";

export interface GrandSubjectColumns {
    Id: Column<GrandSubjectRow>;
    Title: Column<GrandSubjectRow>;
    Description: Column<GrandSubjectRow>;
    CreatedDate: Column<GrandSubjectRow>;
    CreatorUserName: Column<GrandSubjectRow>;
    ModifiedDate: Column<GrandSubjectRow>;
    ModifiedUserName: Column<GrandSubjectRow>;
}

export class GrandSubjectColumns extends ColumnsBase<GrandSubjectRow> {
    static readonly columnsKey = 'GrandSubjectDB.GrandSubject';
    static readonly Fields = fieldsProxy<GrandSubjectColumns>();
}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/GrandSubjectDB/GrandSubject/GrandSubjectColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace CorrespondenceSystem.GrandSubjectDB.Columns;

[ColumnsScript("GrandSubjectDB.GrandSubject")]
[BasedOnRow(typeof(GrandSubjectRow), CheckNames = true)]
public class GrandSubjectColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public Guid Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime CreatedDate { get; set; }
    public string CreatorUserName { get; set; }
    public DateTime ModifiedDate { get; set; }
    public string ModifiedUserName { get; set; }
    //public byte[] TimeStamp { get; set; }
}